pkg/node/types/node.go

```diff
@@ -223,6 +223,22 @@ func (n *Node) getNodeIP(ipv6 bool) (net.IP, addressing.AddressType) {
 	return backupIP, ipType
 }
 
+// GetK8sNodeIPs returns k8s Node IP (either InternalIP or ExternalIP or nil;
+// the former is prefered).
+func (n *Node) GetK8sNodeIP() net.IP {
```

Comment:
I was looking at the file recently and noticed that we have two very similar functions: `GetK8sNodeIP` and `GetNodeIP` (immediately below). I tracked it down to this PR. I am just curious; why did we need to add this @brb?

Reply:
`GetNodeIP()` can return an IP which is neither k8s `InternalIP` nor k8s `ExternalIP`. For the device detection, we want to use k8s IPs, as they are used for communication between nodes. For this, I introduced `GetK8sNodeIP()`.